### PR TITLE
[gatsby-plugin-canonical-urls] Allow forcing a trailing slash

### PR DESCRIPTION
**Pull request by BayPhillips**
This is the last thing I needed to tweak in order to force a trailing slash for my site (a limitation of Github Pages). I also think I fixed a discrepancy in some of the data attributes in the browser (mismatched names).

This shouldn't change any existing behavior as it's entirely opt-in via the new plugin option.

Happy to tweak whatever in the PR as this is the first time I've done it here :)

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Deploy preview for *using-glamor* failed.

Built with commit 1be461f68b44eba29749716045448cac46fa37f0

https://app.netlify.com/sites/using-glamor/deploys/5b4ca5a7b312744d42be279e

File: packages/gatsby-plugin-canonical-urls/src/gatsby-ssr.js [hunk 1]
```diff
@@ -7,14 +7,19 @@ exports.onRenderBody = (
...
     setHeadComponents([
       <link
         rel="canonical"
         key={myUrl}
         href={myUrl}
-        data-baseprotocol={parsedUrl.protocol}
-        data-basehost={parsedUrl.host}
+        data-baseProtocol={parsedUrl.protocol}
+        data-baseHost={parsedUrl.host}
+        data-requireTrailingSlash={`${requireTrailingSlash}`}
```
Comment (pieh) [inline packages/gatsby-plugin-canonical-urls/src/gatsby-ssr.js]: Just thinking out loud here - let me know if this make sense:
Could we perhaps skip adding this attribute if `requireTrailingSlash` isn't true? Not really big deal but would save few bytes for default config and would work the same :)

Comment (BayPhillips) [reply packages/gatsby-plugin-canonical-urls/src/gatsby-ssr.js]: Sounds easy enough, I'll take a stab tonight and update the PR.

**Comment by KyleAMathews** [comment]
Is `data-requireTrailingSlash` a github pages thing? I've never heard of it before.

**Comment by BayPhillips** [comment]
The more I think about it, that should work. The `pathname` passed into `onRenderBody` always going to be the path provided in the initial `createPage` in `gatsby-node.js`, right?

**Comment by KyleAMathews** [comment]
Yeah, Gatsby doesn't touch your path (other than adding an initial slash if it's missing as that's required).

**Comment by BayPhillips** [comment]
@KyleAMathews just double checked this, and you are 100% right (shocking). This PR is unnecessary. Turns out I had already solved this issue for myself when I added this guy: https://github.com/BayPhillips/gatsby-plugin-force-trailing-slashes . I guess I hadn't verified it working with the canonical plugin properly on my live site before jumping into this (I blame Github Pages caching). Closing this, thanks for hearing me out.

**Pull request closed without merging**